pkg/workload/tpcc/payment.go

```diff
@@ -192,11 +208,24 @@ func (p *payment) run(ctx context.Context, wID int) (interface{}, error) {
 		ctx, p.mcp.Get(),
 		func(tx pgx.Tx) error {
 			var wName, dName string
-			// Update warehouse with payment
-			if err := p.updateWarehouse.QueryRowTx(
-				ctx, tx, d.hAmount, wID,
-			).Scan(&wName, &d.wStreet1, &d.wStreet2, &d.wCity, &d.wState, &d.wZip); err != nil {
-				return errors.Wrap(err, "update warehouse failed")
+
+			// When isLongDurationWorkload is true, we want to reset w_ytd
+			if p.config.isLongDurationWorkload && timeutil.Since(p.previousResetTime[wID]) >= warehouseWytdResetPeriod {
+
+				// Reset w_ytd to the initial value.
+				if _, err := p.resetWarehouse.ExecTx(
+					ctx, tx, wYtd, wID,
+				); err != nil {
+					return errors.Wrap(err, "reset warehouse failed")
+				}
+				p.previousResetTime[wID] = timeutil.Now()
```

Comment:
This is not concurrent-safe, as `p` is shared across workers. We'd either want to throw accesses/updates of this map behind a mutex (slow), or alternatively we can use atomics and try to take advantage of the db's txn and `SELECT` for the current value, see if it's high, then `UPDATE` it within the same timestamp. If our transaction succeeds, we can then update the value in the map using atomic operations.

You could also spin up a separate goroutine that waits on ticks from a `time.NewTimer`, and upon a tick it resets the counts of all warehouses within one transaction. That might actually be easier, as we don't need to keep track of `previousResetTime` at all.

Reply:
I initially thought of the second approach itself with a ticket but there can be cases of huge number of warehouses (> 100k), and updating them in a single transaction might be an issue, but that can be solved by creating batches. 

I think since there's state involved here in this implementation with the complexity of concurrency, a different goroutine with a ticker might be better approach. I'll move to that approach now. Thanks for the comment.